packages/components/src/components/date-picker/date-picker-input.jsx

```diff
@@ -37,11 +37,21 @@ const DatePickerInput = ({
         setIsClearBtnVisible(false);
     };
 
+    const OpenIcon = (
+        <DatePickerIcon
+            onClick={onClick}
+            className={classNames({
+                'dc-datepicker__icon': !!onClick, // If has an onclick function add class
+            })}
+            icon='IcCalendar'
+        />
+    );
+
     const trailing_icon =
         is_clearable && is_clear_btn_visible ? (
             <DatePickerIcon icon='IcCloseCircle' onClick={onClickClear} color='secondary' />
         ) : (
-            <DatePickerIcon icon='IcCalendar' />
+            OpenIcon
```

Comment:
Don't we need to wrap this like a component?
```suggestion
            <OpenIcon />
```